mne/preprocessing/maxwell.py

```diff
@@ -321,6 +321,10 @@ def maxwell_filter(raw, origin='auto', int_order=8, ext_order=3,
         sss_ctc = _read_ctc(cross_talk)
         ctc_chs = sss_ctc['proj_items_chs']
         meg_ch_names = [info['ch_names'][p] for p in meg_picks]
+        # checking for extra space ambiguity in channel names
+        # between old and new fif files
+        if meg_ch_names[0][3] != ctc_chs[0][3]:
+            ctc_chs = _clean_names(ctc_chs, remove_whitespace=True)
```

Comment:
Could we also put a `_clean_names` around the list comprehension there, or do we end up with even more problems later? I ask because the check done here isn't robust to e.g. channel reorderings, other systems, future changes, etc.

If adding more `_clean_names` still doesn't do it, maybe just doing `if meg_ch_names[0] not in ctc_chs` is a better check.